### ISSUE TITLE
Object Navigation: Certain Checkboxes Displayed with Numbers Next to Them

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#10739]
### Steps to reproduce:

So far I've only had chance to test this on Windows 7, so can't confirm whether this behaviour can also be reproduced on Windows 10. This is a follow-up to #10681 but with additional info and examples of checkboxes.

1. Open the Folder Options dialog - either directly via the Start Menu, or via the Control Panel

2. Open the "View" tab, and navigate to the list view

3. Place NVDA's focus on any of the checkboxes in the list, and either read the current line or move object navigation to any of the displayed checkboxes

### Actual behavior:

When either using the "read current line" gesture or when focusing NVDA's object navigation on any of the checkboxes in the mentioned list, a number is output by NVDA after the name and state of the checkbox. This behaviour can also be reproduced by going to the Properties dialog for the Start Menu, opening the "Start Menu" tab, hitting the "Customise" button and then performing the same steps with any of the checkboxes displayed in that dialog.

### Expected behavior:

NVDA should only announce the state of the checkbox that has focus - and not the number after it. This number appears to be ignored when using the up/down cursor keys and is only output when reading the current line or when navigating the dialog via object navigation.

**Comment by Adriani90** [comment]
Ok I can reproduce this only with NVDA+l (read current line). I cannot reproduce this with Object navigation. By the way, the object navigation in that tree view is quite broken.
My steps to reproduce:
1. Open Start menu and type folder options or explorer options
2. go to the View tab
3. Press tab until you land on the view settings tree view
4. Press down arrow key to browse the settings
5. press nvda+l (laptop keyboard layout) on any of the settings in the tree view.

Example for the setting "always show checkboxes for selecting elements:
Actual behavior:
Step 4: NVDA says something like "always show checkboxes for selecting elements"
Step 5: NVDA says "always show checkboxes for selecting elements"´,1"

Expected behavior:
When performing step 5 (nvda+l), NVDA should not report the number "1" at the end of the sentence.

cc: @josephsl, @feerrenrut

**Comment by DrSooom** [comment]
@elliott94 and @Adriani90: Start NVDA with the "--no-sr-flag" flag and try to reproduce it. The Windows Explorer window reacts slightly different if a screenreader is running on opening such a window. On my German Win7x64 machines I also get "-OFF" and "-ON" (yes, in English on a German GUI) on such an entry in the dialog window you mentioned. These two suffix values are also visually visible. This behavior is caused by a screenreader (and other similar AT software). I recognized this behavior already in 2010Q2, if I remember correctly.

**Comment by Adriani90** [comment]
Update with NVDA 2024.3 Beta on Windows 11, the number is also displayed on the object level at the end of every object label, when using NVDA+arrow keys to move the review cursor you will notice the numbers on every element.
cc: @josephsl any idea why those numbers are displayed?

You can reproduce this in the explorer options dialog.

**Comment by josephsl** [comment]
Hi,

While I can reproduce the tree view level reporting, I cannot reproduce the situation where NVDA announces position information for checkboxes and radio buttons. Due to incompleteness of the original bug report (please fill out the template next time), I can only guess that this is happening across NVDA releases and on various Windows releases.

Check:

1. Run NVDA with add-ons disabled.
2. Check if "Guess object position information when unavailable" is enabled somehow (object presentation settings panel).

Thanks (postscript: consider me offline throughout the month of July).

**Comment by Adriani90** [comment]
Yes I think the tree view level is displayed as a single character when using review cursor with NVDA+arrow keys (laptop keyboard layout), and this should not be the case. The tree level should be displayed only when navigate between objects, but not when reviewing the characters of an object label. This is how all other tree views work. So it seems this issue is only related to the explorer option dialog, view tab, advanced explorer settings tree view.

**Comment by Adriani90** [comment]
In my case, add-ons are disabled and "Guess object position information when unavailable" is disabled.

**Comment by josephsl** [comment]
Hi,

Now I can reproduce the problem - the original steps and the updated comments weren't clear at first.

For others wondering about this issue, here is a more refined steps to reproduce:

1. Open Start, type "file explorer options", then press Enter.
2. When file explorer options window paears, press Control+Tab to move to view tab.
3. Press Tab until arriving at the options tree view.
4. Move to any checkbox but don't activate it.
5. Use review cursor commands to review checkbox label text.

Expected: NVDA does not include position information as part of the label text while reviewing with the review cursor.
Actual: position information is seen when using review cursor commands to review by character.
Cause: checkbox level/position information (in this case, tree view level) is exposed as object value as shown in the developer info for one of the checkboxes (English UI):

name: 'Use check boxes to select items-OFF'
role: CHECKBOX
processID: 19492
roleText: None
states: SELECTABLE, FOCUSABLE, FOCUSED, SELECTED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.IAccessible.IAccessible object at 0x05348B10>
Python class mro: (<class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: None
location: RectLTWH(left=99, top=362, width=180, height=16)
value: '1'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
appModule: AppModule(explorer, appName='explorer', processID=19492)
appModule.productName: 'Microsoft® Windows® Operating System'
appModule.productVersion: '10.0.26100.1'
appModule.helperLocalBindingHandle: c_long(76266776)
windowHandle: 10162228
windowClassName: 'SysTreeView32'
windowControlID: 30120
windowStyle: 1345388564
extendedWindowStyle: 516
windowThreadID: 25684
windowText: 'Tree1'
displayText: 'Use check boxes to select items-OFF'
IAccessibleObject: <POINTER(IAccessible) ptr=0x75406a0 at 5352210>
IAccessibleChildID: 2
IAccessible event parameters: windowHandle=10162228, objectID=-4, childID=2
IAccessible accName: 'Use check boxes to select items-OFF'
IAccessible accRole: ROLE_SYSTEM_CHECKBUTTON
IAccessible accState: STATE_SYSTEM_SELECTED, STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_SELECTABLE, STATE_SYSTEM_VALID (3145734)
IAccessible accDescription: None
IAccessible accValue: '1'

Thanks.
